Libraries/PushNotificationIOS/RCTPushNotificationManager.m

```diff
@@ -251,4 +251,53 @@ - (void)handleRemoteNotificationsRegistered:(NSNotification *)notification
   }
 }
 
+RCT_EXPORT_METHOD(getScheduledLocalNotifications:(RCTResponseSenderBlock)callback)
+{
+  NSArray<UILocalNotification *> *scheduledLocalNotifications = [UIApplication sharedApplication].scheduledLocalNotifications;
+
+
+  NSMutableArray *formattedScheduledLocalNotifications = [[NSMutableArray alloc] init];
+
+  for (UILocalNotification *notification in scheduledLocalNotifications) {
+
+    NSMutableDictionary* formattedScheduledLocalNotification = [NSMutableDictionary dictionary];
+
+    if (notification.fireDate) {
+      NSDateFormatter *formatter = [[NSDateFormatter alloc] init];
+      [formatter setDateFormat:@"yyyy-MM-dd'T'HH:mm:ss.SSSZZZZZ"];
+      NSString *fireDateString = [formatter stringFromDate:notification.fireDate];
+
+      [formattedScheduledLocalNotification setObject: fireDateString forKey:@"fireDate"];
+    }
+
+    if (notification.alertBody) {
+      [formattedScheduledLocalNotification setObject: notification.alertBody forKey:@"alertBody"];
+    }
+
+    if (notification.alertAction) {
+      [formattedScheduledLocalNotification setObject: notification.alertAction forKey:@"alertAction"];
+    }
+
+    if (notification.soundName) {
+      [formattedScheduledLocalNotification setObject: notification.soundName forKey:@"soundName"];
+    }
+
+    if (notification.category) {
+      [formattedScheduledLocalNotification setObject: notification.category forKey:@"category"];
+    }
+
+    if (notification.applicationIconBadgeNumber) {
+      [formattedScheduledLocalNotification setObject: [NSNumber numberWithInteger:notification.applicationIconBadgeNumber] forKey:@"applicationIconBadgeNumber"];
```

Comment:
Use `@(notification.applicationIconBadgeNumber)` instead of `[NSNumber numberWithInteger:..]`